cityofaustin/atd-moped
# Normalize project status (part 1)

## Associated issues
https://github.com/cityofaustin/atd-data-tech/issues/7910

See also #872, #873.

This PR begins the task of normalizing Moped project status. The ultimate goal is to eliminate the use of the `current_status` and `current_phase` columns on the `moped_project` table, and to entirely remove the `moped_status` table.

Note that although this leaves the Editor in a completely functional state, it's not intended to be merged into main until related subsequent PRs are merged into it.

### Summary of changes
- creates a new view, `current_phase_view` which shows the current phase of all projects that _have a current moped_proj_phase_.
- Establishes hasura relationships between `current_phase_view` and `moped_project`, `moped_phases`, and `moped_proj_phases` - I though this would be useful, but in my subsequent work i'm not sure these relationships are needed...
- updates `project_list_view` to reference the `current_phase_view`
- Reworks the project list config to make use of data from `current_phase_view`. 
- Adds a new `Canceled` and `On hold` phase to `moped_phases`.
- Adds a new column `phase_key` to `moped_phases`.
- Removes references to redundant phase columns on the `moped_project` table: `current_status`, `current_phase`, `current_phase_id`.
- Removes project list references to `moped_project.status_id` - with the intent of dropping the `moped_status` entirely in a subsequent PR
- Removes two editor components we weren't using: `DataTable.js` and `ProjectSummaryTable.js`

## Testing
**URL to test:** Local


**Steps to test:**
1. Open the project list - notice that only five projects have a status badge
2. Edit "Avenue G Healthy Street" (ID 159) - add a construction phase with a start date and make the phase current
3. Confirm the status badge updates accordingly
4. Back in the project list view, notice that the status badge is rendered and the "Construct start" column is populated.


---
#### Ship list
